src/Shared/CopyOnWriteDictionary.cs

```diff
@@ -34,31 +30,20 @@ namespace Microsoft.Build.Collections
     /// be run in a separate appdomain.
     /// </comment>
     [Serializable]
-    internal class CopyOnWriteDictionary<K, V> : IDictionary<K, V>, IDictionary
+    internal class CopyOnWriteDictionary<K, V> : IDictionary<K, V>, IDictionary, ISerializable
     {
-#if DEBUG
-        /// <summary>
-        /// When set forces immediate copy
-        /// </summary>
-        private static readonly bool s_forceWrite = (!String.IsNullOrEmpty(Environment.GetEnvironmentVariable("MSBUILDFORCECOWCOPY")));
-#endif
-
-        /// <summary>
-        /// The default capacity.
-        /// </summary>
-        private readonly int capacity;
-
         /// <summary>
         /// The backing dictionary.
         /// Lazily created.
         /// </summary>
-        private CopyOnWriteBackingDictionary<K, V> backing;
+        private ImmutableDictionary<K, V> backing;
```

Comment:
nit: `_backing` so it's super clear that it's a field. I wouldn't nit-pick on it but here seeing that the write methods mutate the object is critical to understanding the logic.